### PR TITLE
Adds instructions to deploy an Elastic NLP model in an air-gapped environment

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds instructions on how to deploy an NLP model provided by Elastic in an air-gapped environment.

### Preview

[Deploy an Elastic NLP model in an air-gapped environment](https://stack-docs_2370.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-deploy-models.html#ml-nlp-deploy-model-air-gapped)

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 1]
```diff
@@ -157,11 +157,110 @@ change the number of allocations, you can use the
...
+==== Using an HTTP server
+
+INFO: If you use an existing HTTP server, note that the model downloader only 
+supports passwordless HTTP servers.
+
+You can use any HTTP service to deploy a model. This example uses the official 
+Nginx Docker image to set a new HTTP download service up.
 
-[TO DO]
+. Download the model artifact files from https://ml-models.elastic.co/.
```
Comment (hendrikmuhs) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:191]: artifact -> artifacts ?

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 2]
```diff
@@ -157,11 +157,110 @@ change the number of allocations, you can use the
...
++
+--
+[source, shell]
+--------------------------------------------------
+export ELASTIC_ML_MODELS="/path/to/models"
+docker run --rm -d -p 8080:80 --name ml-models -v ${ELASTIC_ML_MODELS}:/usr/share/nginx/html nginx
+--------------------------------------------------
+
+Don't forget to change `/path/to/models` to the path of the subdirectory where 
+the model artifact files are located.
```
Comment (hendrikmuhs) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:203]: artifacts?

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 3]
```diff
@@ -157,11 +157,110 @@ change the number of allocations, you can use the
...
+--
+. Verify that Nginx runs properly by visiting the following URL in your 
+browser:
++
+--
+```
+http://{IP_ADDRESS}:8080/elser_model_1.metadata.json
+```
+
+This example uses the ELSER model artefacts. If you want to work with another
```
Comment (hendrikmuhs) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:218]: art_e_facts -> art_i_facts

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 4]
```diff
@@ -157,11 +157,110 @@ change the number of allocations, you can use the
...
+model provided by Elastic, modify the URL accordingly.
+
+If Nginx runs properly, you see the content of the metdata file of the model.
+--
+. Point your Elasticsearch deployment to the model artifacts on the HTTP server
+by adding the following line to the `config/elasticsearch.yml` file: 
++
+--
+```
+xpack.ml.model_repository: http://{IP_ADDRESS}:8080
```
Comment (hendrikmuhs) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:228]: `{IP_ADDRESS}` -> `{IP_ADDRESS_OR_HOSTNAME}` - we of course support DNS.

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 5]
```diff
@@ -157,11 +157,110 @@ change the number of allocations, you can use the
...
+docker stop ml-models
+--------------------------------------------------
+
+
+[discrete]
+==== Using file-based access
+
+For a file-based access, follow these steps:
+
+. Download the model artifact files from https://ml-models.elastic.co/.
```
Comment (hendrikmuhs) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:250]: artifact -> artifacts?

**Review by hendrikmuhs — COMMENTED** [review]
added some comments.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 6]
```diff
@@ -7,4 +7,107 @@
 :keywords: {ml-init}, {stack}, {nlp}, ELSER
 :description: ELSER is a learned sparse ranking model trained by Elastic.
 
-[TBD]
\ No newline at end of file
+[discrete]
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:10]:
```suggestion
experimental[]

[discrete]
```

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 7]
```diff
@@ -7,4 +7,107 @@
...
+subdirectory containing the model files. As the Docker image has to be 
+downloaded and built, the first start might take a longer period of time. 
+Subsequent runs start quicker.
+--
+. Verify that Nginx runs properly by visiting the following URL in your 
+browser:
++
+--
+```
+http://{IP_ADDRESS}:8080/elser_model_1.metadata.json
```
Comment (hendrikmuhs) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:66]: `IP_ADDRESS_OR_HOSTNAME`

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 8]
```diff
@@ -7,4 +7,109 @@
...
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:66]:
```suggestion
http://{IP_ADDRESS_OR_HOSTNAME}:8080/elser_model_1.metadata.json
```

**Pull request merged**